src/ansys/dynamicreporting/core/utils/enhanced_images.py

```diff
@@ -0,0 +1,482 @@
+    buffer = io.BytesIO()
+    form_multipage_image(json_data, rgb_buffer, pick_buffer, var_buffer, buffer)
+    buffer.seek(0)
+    image = Image.open(buffer)
+    return image
+
+
+def generate_components_for_enhanced_image(
+    model: dpf.Model, var_field: dpf.Field, part_name: str
+) -> Tuple[Dict, np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Esstential helper function for DPF inputs. Generate json metadata, rgb buffer, pick
+    data buffer and variable data buffer from a DPF model object and a DPF field object.
+
+    Parameters
+    ----------
+    model: dpf.Model
+        A DPF model object.
+    var_field: dpf.Field
+        A DPF field object that comes from the given model. The field is essentially
+        the variable in interest to visualize in an enhanced image.
+    part_name: str
+        The name of the part. It will showed on the interactive enhanced image in ADR.
+
+    Returns
+    -------
+    Tuple[Dict, np.ndarray, np.ndarray, np.ndarray]
+        A tuple of JSON metadata, rgb buffer, pick data buffer and variable data buffer
+    """
+    # Todo: vector data support: is_scalar_data = var_data.ndim == 1
+
+    # Get components for metadata
+    var_unit: str = var_field.unit
+    var_name = var_field.name
+    var_meshed_region = var_field.meshed_region
+    dpf_unit_system = model.metadata.result_info.unit_system_name
+    unit_system_to_name = dpf_unit_system.split(":", 1)[0]
+
+    mats: dpf.PropertyField = var_meshed_region.property_field("mat")  # Pick data
+
+    # Convert DPF to a pyvista UnstructuredGrid, which inherits from vtk
+    grid = vtk_helper.dpf_mesh_to_vtk(var_meshed_region)
+    # Add pick data
+    grid = vtk_helper.append_field_to_grid(mats, var_meshed_region, grid, "Pick Data")
+    # Add variable data
+    grid = vtk_helper.append_field_to_grid(var_field, var_meshed_region, grid, var_name)
+
+    # Create a vtkGeometryFilter to convert UnstructuredGrid to PolyData
+    geometry_filter = vtk.vtkGeometryFilter()
+    geometry_filter.SetInputData(grid)
+    geometry_filter.Update()
+    poly_data = geometry_filter.GetOutput()
+
+    renderer, render_window = setup_render_routine(poly_data)
+    rgb_buffer = get_rgb_value(render_window)
+    pick_buffer = render_pick_data(grid, renderer, render_window)
+    var_buffer = render_var_data(grid, renderer, render_window, var_name)
+
+    # Todo: automatic colorby_var support
+    # global colorby_var_id
+    # colorby_var_int = colorby_var_id
+    # colorby_var_id += 1
+    # colorby_var_decimal = 0 if is_scalar_data else 1
+    # Todo: .1, .2, .3 corresponds to x, y, z dimension. Only supports scalar for now
+    # colorby_var = f"{colorby_var_int}.{colorby_var_decimal}"
+
+    # For now, it only supports one part with one variable
+    json_data = {
+        "parts": [
+            {
+                "name": part_name,
+                "id": str(mats.data[0]),
+                "colorby_var": "1.0",  # colorby_var
```

Comment:
The number is hardcoded for now as we only support scalar variable. It can be extended if we want to generate enhanced images for multiple part-var pairs.